docs/examples/driver_examples/calibrate_keithley.py

```diff
@@ -0,0 +1,215 @@
+    for channel in channels:
+        input(f"Please connect channel {channel} to V input on calibrated DMM.")
+        for smu_range, dmm_range in zip(smu_ranges, dmm_ranges):
+            dmm_param.range(dmm_range)
+            calibrate_keithley_smu_v_single(
+                smu, channel, dmm_param.volt, smu_range, src_Z, time_delay
+            )
+    if save_calibrations:
+        save_calibration(smu)
+
+
+def calibrate_keithley_smu_v_single(
+    smu: Instrument,
+    channel: str,
+    dmm_param_volt: Parameter,
+    v_range: str,
+    src_Z: float = 1e-30,
+    time_delay: float = 3.0,
+) -> None:
+    assert channel in channels
+    assert v_range in src_FS_map.keys()
+    src_FS = src_FS_map[v_range]
+
+    if channel == "smua":
+        smun = smu.smua
+    elif channel == "smub":
+        smun = smu.smub
+
+    sense_modes = ["SENSE_LOCAL"]
+
+    for sense_mode in sense_modes:
+        print("Sense mode: " + sense_mode)
+        smu.write(f'{channel}.cal.unlock("KI0026XX")')
+        smu.write(f"{channel}.reset()")
+        smu.write(f"{channel}.source.func = {channel}.OUTPUT_DCVOLTS")
+
+        smu.write(f"{channel}.source.rangev = {v_range}")
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_OFF")
+        smu.write(f"{channel}.sense = {channel}." + sense_mode)
+        time.sleep(time_delay)
+
+        # Start positive calibration:
+        smu.write(f"{channel}.cal.polarity = {channel}.CAL_POSITIVE")
+        smu.write(f"{channel}.source.levelv = {src_Z}")
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_ON")
+
+        # Measure positive zero voltage with SMU and DMM:
+        time.sleep(time_delay)
+        smu.write(f"Z_rdg = {channel}.measure.v()")
+        DMM_Z_rdg = dmm_param_volt()
+
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_OFF")
+        smu.write(f"{channel}.source.levelv = {src_FS}")
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_ON")
+
+        # Measure positive full scale voltage with SMU and DMM:
+        time.sleep(time_delay)
+        smu.write(f"FS_rdg = {channel}.measure.v()")
+        DMM_FS_rdg = dmm_param_volt()
+
+        # Write positive v_range calibration to SMU:
+        smu.write(f"{channel}.source.output = {channel}.OUTPUT_OFF")
+        time.sleep(time_delay)
+        smu.write(
+            f"{channel}.source.calibratev({v_range}, {src_Z}, {DMM_Z_rdg}, {src_FS}, {DMM_FS_rdg})"
+        )
+        if sense_mode != "SENSE_CALA":
+            time.sleep(time_delay)
+            smu.write(
+                f"{channel}.measure.calibratev({v_range}, Z_rdg, {DMM_Z_rdg}, FS_rdg, {DMM_FS_rdg})"
+            )
+        time.sleep(time_delay)
+
+        # Debug output
+        print(
+            f"{channel}.source.calibratev({v_range}, {src_Z}, {DMM_Z_rdg:.6e}, {src_FS:.6f}, {DMM_FS_rdg:.6f})"
+        )
+        print(
+            f"{channel}.measure.calibratev({v_range}, Z_rdg, {DMM_Z_rdg:.6e}, FS_rdg, {DMM_FS_rdg:.6f})"
+        )
+
```

Comment:
these can be removed because qcodes has logging per instrument instance and all commands are logged, so to debug the output it's just a matter of lifting the logging label (see qcodes documentation on logging for details on how to do that)
```suggestion
```

Reply:
Sounds good to me.